backend/pikpak/pikpak.go

```diff
@@ -0,0 +1,1749 @@
+		return nil, err
+	}
+
+	if srcParentID != dstParentID {
+		// Do the move
+		if err = f.moveObjects(ctx, []string{srcObj.id}, dstParentID); err != nil {
+			return nil, err
+		}
+	}
+	dstObj.id = srcObj.id
+
+	var info *api.File
+	if srcLeaf != dstLeaf {
+		// Rename
+		info, err = f.renameObject(ctx, srcObj.id, dstLeaf)
+		if err != nil {
+			return nil, fmt.Errorf("move: couldn't rename moved file: %w", err)
+		}
+	} else {
+		// Update info
+		info, err = f.getFile(ctx, dstObj.id)
+		if err != nil {
+			return nil, fmt.Errorf("move: couldn't update moved file: %w", err)
+		}
+	}
+	return dstObj, dstObj.setMetaData(info)
+}
+
+// copy objects
+func (f *Fs) copyObjects(ctx context.Context, IDs []string, dirID string) (err error) {
+	if len(IDs) == 0 {
+		return nil
+	}
+	req := api.RequestBatch{
+		Ids: IDs,
+		To:  &map[string]string{"parent_id": parentIDForRequest(dirID)},
+	}
+	if err := f.requestBatchAction(ctx, "batchCopy", &req); err != nil {
+		return fmt.Errorf("copy object failed: %w", err)
+	}
+	return nil
+}
+
+// Copy src to this remote using server side copy operations.
+//
+// This is stored with the remote path given.
+//
+// It returns the destination Object and a possible error.
+//
+// Will only be called if src.Fs().Name() == f.Name()
+//
+// If it isn't possible then return fs.ErrorCantCopy
+func (f *Fs) Copy(ctx context.Context, src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debugf(src, "Can't copy - not same remote type")
+		return nil, fs.ErrorCantCopy
+	}
+	err := srcObj.readMetaData(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create temporary object
+	dstObj, dstLeaf, dstParentID, err := f.createObject(ctx, remote, srcObj.modTime, srcObj.size)
+	if err != nil {
+		return nil, err
+	}
+	if srcObj.parent == dstParentID {
+		// api restriction
+		fs.Debugf(src, "Can't copy - same parent")
+		return nil, fs.ErrorCantCopy
+	}
+	// Copy the object
+	if err := f.copyObjects(ctx, []string{srcObj.id}, dstParentID); err != nil {
+		return nil, fmt.Errorf("couldn't copy file: %w", err)
+	}
+
+	// Can't copy and change name in one step so we have to check if we have
+	// the correct name after copy
+	srcLeaf, _, err := srcObj.fs.dirCache.FindPath(ctx, srcObj.remote, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if srcLeaf != dstLeaf {
+		// Rename
+		info, err := f.renameObject(ctx, dstObj.id, dstLeaf)
+		if err != nil {
+			return nil, fmt.Errorf("copy: couldn't rename copied file: %w", err)
+		}
+		err = dstObj.setMetaData(info)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Update info
+		err = dstObj.readMetaData(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("copy: couldn't locate copied file: %w", err)
+		}
+	}
+	return dstObj, nil
+}
+
+// MergeDirs merges the contents of all the directories passed
+// in into the first one and rmdirs the other directories.
+func (f *Fs) MergeDirs(ctx context.Context, dirs []fs.Directory) (err error) {
```

Comment:
Can pikpak have duplicate file names? So the same file name duplicated in the same directory?

If so you need to declare `DuplicateFiles` in Features. If not you don't need this method and you can remove it.